src/nwp_consumer/internal/inputs/noaa/aws.py

```diff
@@ -0,0 +1,312 @@
+        return files
+
+    def mapTemp(self, *, p: pathlib.Path) -> xr.Dataset:  # noqa: D102
+        log.debug(event="mapping raw file to xarray dataset", filepath=p.as_posix())
+
+        # Load the raw file as a dataset
+        try:
+            ds = cfgrib.open_datasets(
+                p.as_posix(),
+            )
+        except Exception as e:
+            log.warn(
+                event="error converting raw file as dataset",
+                error=e,
+                filepath=p.as_posix(),
+            )
+            return xr.Dataset()
+
+        # Process all the parameters into a single file
+        ds = [
+            d for d in ds if any(x in d.coords for x in ["surface", "heightAboveGround", "isobaricInhPa"])
+        ]
+
+        # Split into surface, heightAboveGround, and isobaricInhPa lists
+        surface = [d for d in ds if "surface" in d.coords]
+        heightAboveGround = [d for d in ds if "heightAboveGround" in d.coords]
+        isobaricInhPa = [d for d in ds if "isobaricInhPa" in d.coords]
+
+        # Update name of each data variable based off the attribute GRIB_stepType
+        for i, d in enumerate(surface):
+            for variable in d.data_vars.keys():
+                d = d.rename({variable: f"{variable}_surface_{d[f'{variable}'].attrs['GRIB_stepType']}"})
+            surface[i] = d
+        for i, d in enumerate(heightAboveGround):
+            for variable in d.data_vars.keys():
+                d = d.rename({variable: f"{variable}_{d[f'{variable}'].attrs['GRIB_stepType']}"})
+            heightAboveGround[i] = d
+
+        surface = xr.merge(surface)
+        # Drop unknown data variable
+        surface = surface.drop_vars("unknown_surface_instant")
+        heightAboveGround = xr.merge(heightAboveGround)
+        isobaricInhPa = xr.merge(isobaricInhPa)
+
+        ds = xr.merge([surface, heightAboveGround, isobaricInhPa])
+
+        # Only conform the dataset if requested (defaults to True)
+        if self.conform:
+            # Rename the parameters to the OCF names
+            # Drop variables that are not in the OCF list first
+            ds = ds.drop_vars(
+                names=[v for v in ds.data_vars if v not in PARAMETER_RENAME_MAP.keys()],
+                errors="ignore",
+            )
+            # * Only do so if they exist in the dataset
+            for oldParamName, newParamName in PARAMETER_RENAME_MAP.items():
+                if oldParamName in ds:
+                    ds = ds.rename({oldParamName: newParamName})
+
+            # Delete unwanted coordinates
+            ds = ds.drop_vars(
+                names=[c for c in ds.coords if c not in COORDINATE_ALLOW_LIST],
+                errors="ignore",
+            )
+
+        # * Each chunk is a single time step
+        # Does not use teh "variable" dimension, as this makes a 86GiB dataset for a single timestamp
+        # Keeping variables separate keeps the dataset small enough to fit in memory
+        if self.conform:
```

Comment:
This is what I was thinking of doing for ICON and GDPS as well. So with `conform` all datasets have the `variable` dimension, but without the conform, it keeps them all separate, with the different pressure levels, heights above ground, etc.

Reply:
That sounds good to me!